obspy/clients/filesystem/sds.py

```diff
@@ -0,0 +1,165 @@
+    YEAR   :  4 digit year
+    NET    :  Network code/identifier, up to 8 characters, no spaces
+    STA    :  Station code/identifier, up to 8 characters, no spaces
+    CHAN   :  Channel code/identifier, up to 8 characters, no spaces
+    TYPE   :  1 characters indicating the data type, recommended types are:
+               'D' - Waveform data
+               'E' - Detection data
+               'L' - Log data
+               'T' - Timing data
+               'C' - Calibration data
+               'R' - Response data
+               'O' - Opaque data
+    LOC    :  Location identifier, up to 8 characters, no spaces
+    DAY    :  3 digit day of year, padded with zeros
+
+See https://www.seiscomp3.org/wiki/doc/applications/slarchive/SDS.
+
+:copyright:
+    The ObsPy Development Team (devs@obspy.org)
+:license:
+    GNU Lesser General Public License, Version 3
+    (http://www.gnu.org/copyleft/lesser.html)
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+
+import glob
+import os
+from datetime import timedelta
+
+from obspy import Stream, read
+from obspy.core.util.misc import BAND_CODE
+
+
+SDS_FMTSTR = os.path.join(
+    "{year}", "{network}", "{station}", "{channel}.{type}",
+    "{network}.{station}.{location}.{channel}.{type}.{year}.{doy:03d}")
+
+
+class Client(object):
+    """
+    Request client for SeisComP Data Structure archive on local filesystem.
+
+    For details see the :meth:`~obspy.clients.filesystem.sds.Client.__init__()`
+    method.
+    """
+    def __init__(self, sds_root, sds_type="D", format="MSEED",
+                 fileborder_seconds=30, fileborder_samples=5000):
+        """
+        Initialize a SDS local filesystem client.
+
+        >>> from obspy.clients.filesystem.sds import Client
+        >>> client = Client("/my/SDS/archive/root")  # doctest: +SKIP
+
+        :type sds_root: str
+        :param sds_root: Root directory of SDS archive.
+        :type sds_type: str
+        :param sds_type: SDS data type identifier, one single character. Types
+            recommended by the SDS definition are: 'D' - Waveform data,
+            'E' - Detection data, 'L' - Log data, 'T' - Timing data,
+            'C' - Calibration data, 'R' - Response data, 'O' - Opaque data. Can
+            also be wildcarded by setting to `?` or `*`.
+        :type format: str
+        :param format: File format the data is stored in, see
+            :func:`~obspy.core.stream.read()` for a list of file formats
+            supported by ObsPy. Usually, SDS archives are stored in "MSEED"
+            format. Can be set to `None` for file format autodetection (slowing
+            down the reading).
+        """
+        if not os.path.isdir(sds_root):
+            msg = ("SDS root is not a local directory: " + sds_root)
+            raise IOError(msg)
+        self.sds_root = sds_root
+        self.sds_type = sds_type
```

Comment:
Fair enough. I just think it might be confusing if the same parameter can be set in two places with one overwriting the other but you might be right in that it enables a more convenient use of the client.